doc/string/aref.rdoc

```diff
@@ -0,0 +1,107 @@
+Returns the substring of +self+ specified by the arguments.
+
+Note: The form <tt>self[regexp, capture = 0]</tt>
+updates {Regexp-related global variables}[rdoc-ref:Regexp@Global+Variables];
+the other forms do not.
+
+<b>Form <tt>self[substring]</tt></b>
+
+With string argument +substring+ given,
+returns the matching substring of +self+, if found:
+
+  'hello'['ell']      # => "ell"
+  'hello'['']         # => ""
+  'hello'['nosuch']   # => nil
+  'тест'['ес']        # => "ес"
+  'こんにちは'['んにち'] # => "んにち"
+
+<b>Form <tt>self[regexp, capture = 0]</tt></b>
+
+With Regexp argument +regexp+ given and +capture+ as zero,
+searches for a matching substring in +self+;
+updates {Regexp-related global variables}[rdoc-ref:Regexp@Global+Variables]:
+
+  'hello'[/ell/]     # => "ell"
+  [$`, $&, $']       # => ["h", "ell", "o"]
+  'hello'[/l+/]      # => "ll"
+  [$`, $&, $']       # => ["he", "ll", "o"]
+  'hello'[//]        # => ""
+  [$`, $&, $']       # => ["", "", "hello"]
+  'hello'[/nosuch/]  # => nil
+  [$`, $&, $']       # => [nil, nil, nil]
```

Comment:
What's the significance of showing the regexp global variables? It seems just follow regular regexp matching behavior.

Reply:
I was thinking to remind readers how useful these globals are.  Dr. Johnson tells us, “People need to be reminded more often than they need to be instructed.”  Will remove if you think it best.